### DISCUSSION TITLE
How to Log  DNS respons in CoreDNS

### DISCUSSION THREAD

**Discussion by YogevMaty** [coredns/coredns#6923] [Q&A]
Hi CoreDNS community,

I'm currently using the log plugin to capture DNS query logs, but I need to extend this functionality to log the resolved IP addresses from DNS responses. I understand that the log plugin can capture query details (e.g., {type}, {name}, {rcode}), but it doesn't seem to have a placeholder for logging the resolved IP addresses directly.

Is there a built-in way or a recommended approach to log the IP addresses that are returned in DNS responses to stdout? 

Any guidance or suggestions would be appreciated!

Thanks in advance.

**Comment by chrisohaver** [comment]
Logging answers isn’t available in the log plugin , but you could use the dnstap plugin with the dnstap CLI tool to write the responses to stdout. I think there’s an example in the dnstap plugin readme.
